query-coordinator/src/main/scala/com/socrata/querycoordinator/resources/QueryService.scala

```diff
@@ -183,6 +183,10 @@ trait QueryService {
     GatewayTimeout
   }
 
+  def ranOutOfRetriesResponse: HttpResponse = {
+    internalServerError
```

Comment:
> 
> 10.5.3 502 Bad Gateway
> 
> The server, while acting as a gateway or proxy, received an invalid response from the upstream server it accessed in attempting to fulfill the request.
> 
> 10.5.4 503 Service Unavailable
> 
> The server is currently unable to handle the request due to a temporary overloading or maintenance of the server. The implication is that this is a temporary condition which will be alleviated after some delay. If known, the length of the delay MAY be indicated in a Retry-After header. If no Retry-After is given, the client SHOULD handle the response as it would for a 500 response.
> 
>       Note: The existence of the 503 status code does not imply that a
>       server must use it when becoming overloaded. Some servers may wish
>       to simply refuse the connection.
> 10.5.5 504 Gateway Timeout
> 
> The server, while acting as a gateway or proxy, did not receive a timely response from the upstream server specified by the URI (e.g. HTTP, FTP, LDAP) or some other auxiliary server (e.g. DNS) it needed to access in attempting to complete the request.
> 
>       Note: Note to implementors: some deployed proxies are known to
>       return 400 or 500 when DNS lookups time out.
> 


I think I'd be fine with 502, 503 or 504... I consider "can't connect" to be a form of a timeout.

My main concern is that it is *not* an internal server error because it isn't an unexpected condition... if anything it is an external server error.  You in fact just wrote all the code to make it a not unexpected condition that we realize isn't QC's fault and it has done what it can to try to retry.  :)  Putting the blame in the right place in the response makes production metrics and monitoring much easier.